### PR TITLE
[jit] Convenience APIs for script objects

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#20226 [jit] Convenience APIs for script objects**

The compiler should be emitting static lookups to object slots, but
people using the C++ API should have some sugar for doing attribute
lookup.

Differential Revision: [D15243625](https://our.internmc.facebook.com/intern/diff/D15243625)

### CODE REVIEW

File: aten/src/ATen/core/ivalue.cpp [hunk 1]
```diff
@@ -113,6 +113,16 @@ std::string ivalue::Object::name() const {
   return this->type_->qualname();
 }
 
+IValue ivalue::Object::getAttr(const std::string& name) const {
```
Comment (zdevito) [inline aten/src/ATen/core/ivalue.cpp]: Probably want the same api as pybind11, which uses `attr`

File: test/cpp/jit/test_class_import.h [hunk 2]
```diff
@@ -58,6 +63,32 @@ void testClassImport() {
...
+      constantTable,
+      nullptr);
+  import_libs(
+      m2.class_compilation_unit(),
+      "__torch__",
+      classSrcs2,
+      constantTable,
+      nullptr);
+
+  // Incorrect arguments for constructor should throow
```
Comment (zdevito) [inline test/cpp/jit/test_class_import.h]: throooooow

**Review by zdevito — APPROVED** [review]

File: test/cpp/jit/test_class_import.h (hunk 2 above)
Comment (suo) [reply test/cpp/jit/test_class_import.h]: damn the macbook keyboard :(

File: aten/src/ATen/core/ivalue.cpp (hunk 1 above)
Comment (suo) [reply aten/src/ATen/core/ivalue.cpp]: pybind's attr returns a functor which lets you write to the object by assigning to it. I think that's probably a little overkill in this case.

I think it'll expose static `getattr` and `setattr` like the python builtins

Comment (TheCodez) [reply aten/src/ATen/core/ivalue.cpp]: The old API seems simpler, this makes it more like a C API. Maybe we shouldn't try to match pythons semantics here?

Comment (suo) [reply aten/src/ATen/core/ivalue.cpp]: I'm okay with that. whatever is more natural for people I guess

**Pull request closed without merging**